### DISCUSSION TITLE
Updates/roadmap for development

### DISCUSSION THREAD

**Discussion by enrico223** [ahrm/sioyek#734] [Q&A] [answered]
Hi @ahrm, thanks for this fantastic reader! Just wanted to know, do you have a place where you update on the development progress? Or just a roadmap for future development? I tried to look around, but didn't find anything

**Comment by ahrm** [comment, marked as answer]
There is no such place but I guess I'll provide an update here:

I am working on the next version of sioyek, because it is a drastic change it doesn't work with the current build system so I decided not to push changes until I have updated the build system to work with the new source code. The major new change is touch mode which allows sioyek to work on touch tablets, which means sioyek can be ported to android and iOS. The android port is done but I am having some trouble porting it to IOS (this is what is holding the release back), but I am confident the issues with IOS can be solved. Here are some of the new improvements that will be available in next release:

* Android and iOS version
* Touch mode
* Freehand drawing annotations
* Built-in feature to search and download papers (no need to install extensions)
* Built-in text to speech
* Rewrote the command system to allow more complex keybindings (for example a key can mean different commands in different contexts)
* And much more
* Updated Qt from 5.14 to 6.5 . This should automatically fix some bugs, for example the titlebar in windows should respect the dark mode and some sizing issues on MacOS should be fixed.
* Updated mupdf to 1.22 which now supports page labels

**Reply by viccie30** [reply]
That sounds like some great new changes! Could you give me a ping when you've pushed the build system changes to see if I can make them work in Debian?

**Reply by ahrm** [reply]
Absolutely.

**Reply by enrico223** [reply]
Hey, are there any updates on development?

**Reply by ahrm** [reply]
I am waiting for this bug to be fixed: https://bugreports.qt.io/browse/QTBUG-114363 .

**Reply by zetashift** [reply]
Since that is iOS only, is there a way to test the Android version? I would really like cross platform annotations + offline reading!

**Reply by ahrm** [reply]
The development branch of this repo can be built for android.

**Reply by zetashift** [reply]
Ohh, that's amazing! Thank you very much :D

**Reply by embie27** [reply]
Are there any build instruction to build the android app? I don't know much about qt on android and I couldn't figure it out.

**Reply by zetashift** [reply]
I'm also interested in build instructions for the android app, even a simple `apk` for download would be ideal, but I'm not sure how to get started with compiling Android binaries for Sioyek

**Comment by enrico223** [comment]
WOW, this update is huge! Do you have a month in mind for when you're going  to release it?

**Reply by ahrm** [reply]
Unfortunately, I can't because the issues with the iOS version are new territory for me so I can't give a precise timeline.

**Reply by hasantahir** [reply]
Wow! Eagerly looking forward to an iOS version. On a different note, it would be great if you can kindly make available the latest build [at hexomancer](https://github.com/hexomancer/sioyek/releases/). I have never been able to successfully compile on a Mac. 

Many thanks for the amazing work!
